File: BAEKJOON/Python/1000/1400/1417.py
"""
Title : 국회의원 선거
Link : https://www.acmicpc.net/problem/1417
"""

from sys import stdin

input = stdin.readline
II = lambda: int(input())

if __name__ == "__main__":
    N = II()
    dasom = II()
    votes = [0] * 101
    for _ in range(N - 1):
        v = II()
        votes[v] += 1
    answer = 0
    for i in range(100, 0, -1):
        if not votes[i]:
            continue
        while votes[i]:
            if dasom > i:
                break
            dasom += 1
            votes[i] -= 1
            votes[i - 1] += 1
            answer += 1
        if (not votes[i] and dasom >= i) or dasom > i:
            break
    print(answer)

File: BAEKJOON/Javascript/1000/1400/1417.js
const fs = require("fs")
const filepath = process.platform === "linux" ? "/dev/stdin" : "./input.txt"
const input = fs.readFileSync(filepath).toString().trim().split("\n")

const N = parseInt(input[0])
let dasom = parseInt(input[1])
let votes = Array(101).fill(0)

for (let i = 0; i < N - 1; i++) {
  v = parseInt(input[i + 2])
  votes[v]++
}

let answer = 0
for (let i = 100; i > 0; i--) {
  if (votes[i] === 0) {
    continue
  }
  while (votes[i] > 0) {
    if (dasom > i) {
      break
    }
    dasom++
    votes[i]--
    votes[i - 1]++
    answer++
  }
  if ((votes[i] !== 0 && dasom >= i) || dasom > i) {
    break
  }
}
console.log(answer)
